Code/Mantid/MantidQt/CustomInterfaces/src/Muon/ALCDataLoadingPresenter.cpp

```diff
@@ -51,19 +52,42 @@ namespace CustomInterfaces
         alg->setProperty("TimeMin", timeRange->first);
         alg->setProperty("TimeMax", timeRange->second);
       }
+
+      // If corrections from custom file requested, set file property
       if (m_view->deadTimeType() == "FromSpecifiedFile") {
         alg->setProperty("DeadTimeCorrFile",m_view->deadTimeFile());
       }
 
+      // If custom grouping requested, set forward/backward groupings
+      if ( m_view->detectorGroupingType() == "Custom" ) {
+        alg->setProperty("ForwardSpectra",m_view->getForwardGrouping());
+        alg->setProperty("BackwardSpectra",m_view->getBackwardGrouping());
+      }
+
+      // If Subtract checkbox is selected, set green period
+      if ( m_view->subtractIsChecked() ) {
+        alg->setProperty("Green",m_view->greenPeriod());
+      }
+
       alg->setPropertyValue("OutputWorkspace", "__NotUsed");
       alg->execute();
 
       m_loadedData = alg->getProperty("OutputWorkspace");
 
-      assert(m_loadedData); // If errors are properly caught, shouldn't happen
-      assert(m_loadedData->getNumberHistograms() == 1); // PlotAsymmetryByLogValue guarantees that
+      // If errors are properly caught, shouldn't happen
+      assert(m_loadedData);
```

Comment:
A comment for future/ongoing work with this ACL interface: maybe these asserts could be replaced with proper exception throws. That would show a bad (but still more user friendly) message to the user, rather than an assert crash.